pm_uep_plot/pm_uep_plot.xml

```diff
@@ -0,0 +1,82 @@
+<tool id="pm_uep_plot" name="generate plots from charge density" version="0.1.1" python_template_version="3.5">
+    <requirements>
+        <requirement type="package">pymuonsuite</requirement>
+        <requirement type="package">zip</requirement>
+    </requirements>
+    <command detect_errors="exit_code"><![CDATA[
+        ln -s $charge_density $charge_density.name &&
+        x="$charge_density.name" &&
+        echo "calculated seed name \$x" &&
+        ln -s $castep_log "\${x%%.*}.castep" &&
+        touch "\${x%%.*}.yaml" &&
+        printf "chden_path: ./ \n" >>"\${x%%.*}.yaml" &&
+        ([[ ! -z "$chden_seed" ]] && printf "chden_seed: $chden_seed \n">>"\${x%%.*}.yaml" || ( >&2 echo "chden_seed empty")) &&
+        ([[ ! -z "$gw_factor" ]] && printf "gw_factor: $gw_factor \n">>"\${x%%.*}.yaml" || ( >&2 echo "gw_factor empty")) &&
+        ([[ ! -z "$line_plots" ]] && ( printf "line_plots: $line_plots \n" | sed "s/__ob__/[/g" | sed "s/__cb__/]/g" )>>"\${x%%.*}.yaml" || ( >&2 echo "line_plots empty" && exit 2)) &&
+        ([[ ! -z "$plane_plots" ]] && ( printf "plane_plots: $plane_plots \n" | sed "s/__ob__/[/g" | sed "s/__cb__/]/g" )>>"\${x%%.*}.yaml" || ( >&2 echo "plane_plots empty" && exit 2)) &&
+        cat "\${x%%.*}.yaml" &&
+        pm-uep-plot "\${x%%.*}.yaml" &&
+        ls > tree.txt &&
+        zip out.zip * -x "tree.txt"
+    ]]></command>
+    <inputs>
+        <param type="data" name="charge_density" label="Charge density file (.den_fmt)" format="txt" help="The charge density file created by CASTEP during your initial DFT simulation for the structure."/>
+        <param type="data" name="castep_log" label="CASTEP log (.castep)" format="txt" help="The CASTEP log for your initial DFT simulation for the structure."/>
+        <param type="text" name="line_plots" label="Line plot coordinates" value="[]" help="Specifications for paths. Check help section for possible formats."/>
+        <param type="float" name="gw_factor" value="5.0" label="Gaussian witdh factor" help="Gaussian width factor for UEP calculation. Higher values will make the potential of atomic nuclei closer to the point-like approximation but may introduce artifacts."/>
+        <param type="text" name="chden_seed" label="charge density seed" value="" help="Seedname for the charge density calculation"/>
```

Comment:
Integration test: when I set chden_seed to something other than the seed for the den_fmt and castep files, the job fails. If I set it to match the den_fmt and castep files, it's superfluous anyway since it's already calculated in the code. I don't think this parameter is necessary to include.

Reply:
replaced with autogenerated seed